### DISCUSSION TITLE
Can I use both SSO and local 2FA?

### DISCUSSION THREAD

**Discussion by kabachook** [gravitational/teleport#9137] [Q&A] [answered]
I would like to use Per-session MFA with ADFS, however:
1. [Per-Session MFA docs page](https://goteleport.com/docs/ver/8.0/access-controls/guides/per-session-mfa/) states that: 
> Per-session MFA checks **don't apply to regular Teleport logins** (tsh login or logging into the Web UI). We encourage you to **enable login MFA in your SSO provider** and/or for all local Teleport users.

and requires
> WebAuthn configured on this cluster
2. [Authentication docs page](https://goteleport.com/docs/ver/8.0/setup/reference/authentication/#saml) states that:
> SSO users can also register 2FA devices, **but Teleport will not prompt them for 2FA during login**. Login 2FA for SSO users should be handled by the SSO provider.

So I can't use both Webauthn + SSO for per-session MFA?
I would love to have this feature, since requiring 2FA for each ssh session is a great way to secure critical infrastructure

**Comment by klizhentas** [comment, marked as answer]
@kabachook Per session MFA works with SSO logins.  I've sent a PR to remove confusion

https://github.com/gravitational/teleport/pull/9140
